internal/github/profile_test.go

```diff
@@ -0,0 +1,315 @@
+    - name: "buildkite-plugin"
+      # array of repos accessible to the profile
+      repositories: 
+        - deploy-templates-buildkite-plugin
+        - very-private-buildkite-plugin
+      permissions: ["contents:read"]
+      
+    # allow package access to any repository
+    - name: "package-registry"
+    # '*' indicates all, when specified must be only value. No other wildcards supported.
+      repositories: ["*"]
+      permissions: ["packages:read"]
+`
+
+	_, err := github.ValidateProfile(context.Background(), profile)
+
+	require.NoError(t, err)
+}
+
+// Test case where the profile that is loaded is invalid
+func TestInvalidProfile(t *testing.T) {
+	// Example of invalid profile content
+	profile := `
+organisation:
+  profilez:
+    # allow read access to a set of buildkite-plugins
+    - name: "buildkite-plugin"
+      # array of repos accessible to the profile
+      repositories: 
+        - deploy-templates-buildkite-plugin
+        - very-private-buildkite-plugin
+      permissions: ["contents:read"]
+      
+    # allow package access to any repository
+    - name: "package-registry"
+      # '*' indicates all, when specified must be only value. No other wildcards supported.
+      repositories: ["*"]
+      permissions: ["packages:read"]
+`
+
+	_, err := github.ValidateProfile(context.Background(), profile)
+
+	require.Error(t, err)
+
+}
+
+func TestLoadProfile(t *testing.T) {
+	profile := `
+organization:
+  profiles:
+    # allow read access to a set of buildkite-plugins
+    - name: "buildkite-plugin"
+      # array of repos accessible to the profile
+      repositories: 
+        - deploy-templates-buildkite-plugin
+        - very-private-buildkite-plugin
+      permissions: ["contents:read"]
+      
+    # allow package access to any repository
+    - name: "package-registry"
+    # '*' indicates all, when specified must be only value. No other wildcards supported.
+      repositories: ["*"]
+      permissions: ["packages:read"]
+`
+
+	router := http.NewServeMux()
+
+	router.HandleFunc("/repos/chinmina/chinmina-bridge/contents/docs/profile.yaml", func(w http.ResponseWriter, r *http.Request) {
+
+		JSON(w, &api.RepositoryContent{
+			Content: &profile,
+		})
+	})
+
+	svr := httptest.NewServer(router)
+	defer svr.Close()
+
+	// generate valid key for testing
+	key := generateKey(t)
+
+	// Example of a valid profile URL
+	configURL := "github.com/chinmina/chinmina-bridge/docs/profile.yaml"
+	gh, err := github.New(
+		context.Background(),
+		config.GithubConfig{
+			ApiURL:         svr.URL,
+			PrivateKey:     key,
+			ApplicationID:  10,
+			InstallationID: 20,
+		},
+	)
+	require.NoError(t, err)
+
+	err = github.LoadProfile(context.Background(), gh, configURL)
+	require.NoError(t, err)
+}
+
+// Test the case where the profile is inconsistent with the request made to Chinmina
+// In this case, the target repository is not included in the targeted profile
+func TestInconsistentProfile(t *testing.T) {
```

Comment:
Consider a single data test for the following three tests, as they only differ in their assertions AFAICT